### PR TITLE
New package: kdepim-addons 21.08.3

### PR DESCRIPTION
**Pull request by DBLouis**
#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I will use the affected packages and briefly tested this PR

### CODE REVIEW

**Comment by amak79** [comment]
I think you need `messagelib-devel` for `KF5WebEngineViewer`.

**Comment by DBLouis** [comment]
Thank you. How did you find out?

**Comment by Johnnynator** [comment]
You can just introduce multiple packages in a single PR if they depend on each other. Please rwbase the PR on top of master.

**Comment by DBLouis** [comment]
Any ideas on how to fix the error on ARM platforms?

` tools/generate_geocoding_data: tools/generate_geocoding_data: cannot execute binary file`

```
file masterdir/builddir/libphonenumber-8.12.23/build/tools/generate_geocoding_data
[...] ELF 64-bit LSB pie executable, ARM aarch64, version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, BuildID [...], for GNU/Linux 3.7.0, not stripped
```
In chroot:

```
ldd builddir/libphonenumber-8.12.23/build/tools/generate_geocoding_data
	not a dynamic executable
```

**Comment by Johnnynator** [comment]
> Any ideas on how to fix the error on ARM platforms?
> 
> ` tools/generate_geocoding_data: tools/generate_geocoding_data: cannot execute binary file`
> 
> ```
> file masterdir/builddir/libphonenumber-8.12.23/build/tools/generate_geocoding_data
> [...] ELF 64-bit LSB pie executable, ARM aarch64, version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, BuildID [...], for GNU/Linux 3.7.0, not stripped
> ```
> 
> In chroot:
> 
> ```
> ldd builddir/libphonenumber-8.12.23/build/tools/generate_geocoding_data
> 	not a dynamic executable
> ```

`build_helper="qemu"`

**Comment by DBLouis** [comment]
It is still failing with the exact same error

File: srcpkgs/libphonenumber/template [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+makedepends="boost-devel gtest-devel icu-devel protobuf-devel"
+short_desc="Google's common library for handling international phone numbers"
+maintainer="Louis Dupré Bertoni <contact@louisdb.xyz>"
+license="Apache-2.0"
+homepage="https://github.com/googlei18n/libphonenumber"
+distfiles="https://github.com/google/libphonenumber/archive/refs/tags/v${version}.tar.gz"
+checksum=49aed4e11eecff703207ad46f40cdb5995eece24d0f7e9aa25537aef5f011e51
+
+do_configure() {
+	cmake -S cpp -B build \
```
Comment (Johnnynator) [inline srcpkgs/libphonenumber/template:16]: Do not override do_configure/do_build/do_install/do_check, you can add what is needed via configure_args ( but you probably only need to set `build_wrksrc=cpp` )

**Review by Johnnynator — CHANGES_REQUESTED** [review]

File: srcpkgs/libphonenumber/template (hunk 1 above)
Comment (DBLouis) [reply srcpkgs/libphonenumber/template:16]: I originally tried it like that but it fails:

```
CMake Error:
  Running

   '/usr/bin/ninja' '-C' '/builddir/libphonenumber-8.12.23/cpp/build' '-t' 'recompact'

  failed with:

   ninja: error: build.ninja:1433: multiple rules generate ../src/phonenumbers/metadata.h [-w dupbuild=err]

CMake Generate step failed.  Build files cannot be regenerated correctly.
```

Comment (Johnnynator) [reply srcpkgs/libphonenumber/template:16]: set `make_cmd=make` and `export CMAKE_GENERATOR="Unix Makefiles"`

Comment (DBLouis) [reply srcpkgs/libphonenumber/template]: thanks for the help :)

File: srcpkgs/kdepim-addons/template [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+configure_args="-DBUILD_TESTING=OFF"
+hostmakedepends="extra-cmake-modules python3 kdoctools kcoreaddons kaddressbook
+ gettext qt5-qmake qt5-host-tools"
+makedepends="akonadi-import-wizard-devel discount-devel incidenceeditor-devel
+ kontactinterface-devel kitinerary-devel ktnef-devel libksieve-devel mailcommon-devel
+ mailimporter-devel messagelib-devel poppler-qt5-devel qt5-devel kpkpass-devel"
+short_desc="Addons for KDE PIM applications"
+maintainer="Louis Dupré Bertoni <contact@louis.xyz>"
+license="GPL-2.0-or-later, GPL-3.0-or-later, LGPL-2.1-or-later, AGPL-3.0-or-later, BSD-3-Clause, BSD-2-Clause"
+homepage="https://invent.kde.org/unmaintained/kdepim"
```
Comment (Chocimier) [inline srcpkgs/kdepim-addons/template:15]: https://invent.kde.org/pim/kdepim-addons

File: srcpkgs/kitinerary/template [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Template file for 'kitinerary'
+pkgname=kitinerary
+version=21.04.0
+revision=1
+build_style=cmake
+configure_args="-DBUILD_TESTING=OFF"
```
Comment (Chocimier) [inline srcpkgs/kitinerary/template:6]: Upstream may want to know about failing tests.

Comment (DBLouis) [reply srcpkgs/kitinerary/template:6]: It fails now because I think extended tests need a X server running.

Comment (paper42) [reply srcpkgs/kitinerary/template:6]: Did you try running the tests with xvfb-run?

Comment (Chocimier) [reply srcpkgs/kitinerary/template:6]: Qt do not need X server with `pre_check() {export QT_QPA_PLATFORM=offscreen;}`, and then they still fail.

Comment (DBLouis) [reply srcpkgs/kitinerary/template:6]: What should I do then?

Comment (Chocimier) [reply srcpkgs/kitinerary/template:6]: Best to work it out with upstream, things like below have potential to make people sad.

```diff
@@ -209,11 +181,7 @@
             },
             "departureDay": "2016-06-10",
             "departureTerminal": "2F",
-            "departureTime": {
-                "@type": "QDateTime",
-                "@value": "2016-06-10T13:00:00+02:00",
-                "timezone": "Europe/Paris"
-            },
+            "departureTime": "2016-06-10T13:00:00",
             "flightNumber": "8680"
         },
         "reservationNumber": "123456",
FAIL!  : ExtractorTest::testExtract(amadeus-amadeus_1.txt) Compared values are not the same
```

Comment (DBLouis) [reply srcpkgs/kitinerary/template:6]: https://invent.kde.org/pim/kdepim-addons/-/issues/1

Comment (Chocimier) [reply srcpkgs/kitinerary/template:6]: I was talking about kitinerary tests up to now, but yes, kdepim-addons has some meaningful failues too, eg.

```
 30/163 Test  #30: pimeventsplugintest ................................................***Failed    0.10 sec
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-piotr'
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-piotr'
********* Start testing of PimEventsPluginTest *********
Config: Using QtTest library 5.15.2, Qt 5.15.2 (x86_64-little_endian-lp64 shared (dynamic) release build; by GCC 10.2.1 20201203), unknown unknown
PASS   : PimEventsPluginTest::initTestCase()
QWARN  : PimEventsPluginTest::testLoadEventsForDataRange() kf.calendarcore: Invalid dtStart
QWARN  : PimEventsPluginTest::testLoadEventsForDataRange() kf.calendarcore: Invalid dtStart
QWARN  : PimEventsPluginTest::testLoadEventsForDataRange() kf.calendarcore: Invalid dtStart
QWARN  : PimEventsPluginTest::testLoadEventsForDataRange() kf.calendarcore: Invalid dtStart
FAIL!  : PimEventsPluginTest::testLoadEventsForDataRange() Compared values are not the same
   Actual   (actual.size())  : 2
   Expected (expected.size()): 40
```

On dbus: runing tests under dbus-run-session like okular template does does not really help - test 2 freeze (but it passed without dbus - ???). Maybe we would need to disable dbus depending test after deciding to ignore/resolving others.

**Comment by DBLouis** [comment]
I think it does not run the same tests on my machine, I never got those one failing

File: srcpkgs/kitinerary/template [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ kpkpass-devel libphonenumber-devel poppler-devel protobuf-devel qt5-devel"
+short_desc="Data model and extraction system for travel reservation information"
+maintainer="Louis Dupré Bertoni <contact@louisdb.xyz>"
+license="LGPL-2.1-or-later"
+homepage="https://kontact.kde.org"
+distfiles="${KDE_SITE}/release-service/${version}/src/${pkgname}-${version}.tar.xz"
+checksum=6e7d052888b4e8f4b8b81a34917204c6d2326e10af1f56dcaf39d0f204c1d918
+
+if [ "$CROSS_BUILD" ]; then
+	hostmakedepends+=" kcoreaddons qt5-host-tools qt5-qmake qt5-tools-devel"
```
Comment (Johnnynator) [inline srcpkgs/kitinerary/template:17]: You can just keep them unconditional

**Comment by Johnnynator** [comment]
Can you make a commit per touched template? Cross and musl still seem to be broken.

**Comment by DBLouis** [comment]
> Cross and musl still seem to be broken.

It looks like it fail to compile on those I don't understand why

**Review by Johnnynator — APPROVED** [review]

**Comment by Johnnynator** [comment]
I did take a look at libphonenumber. Something is really wonky there. If I compile with `-j3` it fails half the time or more. With `-j32` it never fails. There is most likely a race condition in the Makefile. Importing the patch alpine use to get ninja working seems to not hit this issue. I added an update for `libphonenumber` + patch to your branch.

I will merge this together with some more kde updates to `21.12.0` Sunday / earlier next week.

**Pull request closed without merging**